stats/system_tag.go

```diff
@@ -0,0 +1,133 @@
+package stats
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+)
+
+// SystemTagSet is a bitmask that is used to keep track
+// which system tags should be included with which metrics.
+type SystemTagSet uint32
+
+// TagSet is a string to bool map (for lookup efficiency) that is used to keep track
+// which system tags and non-system tags should be included with with metrics.
+type TagSet map[string]bool
+
+//nolint: golint
+const (
+	// Default system tags includes all of the system tags emitted with metrics by default.
+	TagProto SystemTagSet = 1 << iota
+	TagSubProto
+	TagStatus
+	TagMethod
+	TagURL
+	TagName
+	TagGroup
+	TagCheck
+	TagError
+	TagErrorCode
+	TagTLSVersion
+
+	// System tags not enabled by default.
+	TagIter
+	TagVU
+	TagOCSPStatus
+	TagIP
+)
+
+// DefaultSystemTagSet includes all of the system tags emitted with metrics by default.
+// Other tags that are not enabled by default include: iter, vu, ocsp_status, ip
+//nolint:gochecknoglobals
+var DefaultSystemTagSet = TagProto | TagSubProto | TagStatus | TagMethod | TagURL | TagName | TagGroup |
+	TagCheck | TagCheck | TagError | TagErrorCode | TagTLSVersion
+
+// ToSystemTagSet converts a tag set to tag set
+// Non-system tag will be ignored.
+func (ts TagSet) ToSystemTagSet() SystemTagSet {
```

Comment:
Why is this actually required? I can't think of a valid use case for this method - any non-system user specified metric tags would be discarded.

Reply:
Yes, I'm totally confused between `SystemTagSet` and `TagSet`, because previous code use the same `TagSet` for both. I understand now.